modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/api/ToolingService.java

```diff
@@ -17,8 +22,24 @@
    * Provides a service to create a connectivity testing service using a builder which can be used to configured resources of a
    * dynamically created artifact
    *
-   * @return a builder to create a {@link org.mule.runtime.module.tooling.api.connectivity.ConnectivityTestingService}
+   * @return a builder to create a {@link ConnectivityTestingService}
    */
   ConnectivityTestingServiceBuilder newConnectivityTestingServiceBuilder();
 
+  /**
+   * Creates an {@link Application} from a set of resources.
+   *
+   * The created application will be created lazily meaning that the application resources
+   * will be created based on the different request made to the application.
+   *
+   * Only requested components will be executed. All sources for flows will be stop unless
+   * they are requested to be started by the client.
+   *
+   * @param applicationLocation location of the application content. The application content
+   *                            may be a folder holding an exploded structure for an application
+   *                            or may be a zip file containing the resources of the application.
+   * @return the created application.
+   */
+  Application createApplication(File applicationLocation) throws IOException;
```

Comment:
The idea is that the ToolingService is the one that can be injected to a core extension (Agent) so we centralize all the required functionality for tooling there. But I agree with your point. We can probably keep using ToolingService as centralization point but it can provide another service that let's you deal with those use cases instead of just exposing createApplication.